contracts/DxMgnPool.sol

```diff
@@ -124,26 +130,38 @@ contract DxMgnPool is Ownable {
     }
 
     function triggerMGNunlockAndClaimTokens() public {
-        require(currentState() == State.PoolingEnded, "Pooling period is not yet over.");
+        checkForStateUpdate();
+        require(currentState == State.PoolingEnded, "Pooling period is not yet over.");
         require(
             dx.getAuctionIndex(address(depositToken), address(secondaryToken)) > lastParticipatedAuctionIndex, 
             "Last auction is still running"
         );      
-        
+
         (address sellToken, address buyToken) = buyAndSellToken();
         address(dx).call(abi.encodeWithSignature("claimSellerFunds(address,address,address,uint256)", buyToken, sellToken, address(this), lastParticipatedAuctionIndex));
         mgnToken.unlockTokens();
         totalDeposit = dx.balances(address(depositToken), address(this));
         if(totalDeposit > 0){
             dx.withdraw(address(depositToken), totalDeposit);
         }
+        currentState = State.DepositWithdrawnFromDx;
     }
 
     function withdrawUnlockedMagnoliaFromDx() public {
-        require(currentState() == State.PoolingEnded, "Pooling period is not yet over.");
-        
-        mgnToken.withdrawUnlockedTokens();
+        require(currentState == State.DepositWithdrawnFromDx, "Unlocking not yet triggered");
+
+        // Implicit we also have:
+        // require(unlockedTokens[msg.sender].withdrawalTime < now, "The tokens cannot be withdrawn yet");
+
         totalMgn = mgnToken.balanceOf(address(this));
+        mgnToken.withdrawUnlockedTokens();
```

Comment:
I'm not sure you can invert the order here. As far as I understand the FRTToken contract, we don't add tokens to the balance before `withdrawUnlockedTokens`, thus `balanceOf` would likely be 0 before.

Reply:
Uh this slipped in. Thanks a lot for the careful review!

Why do my tests still pass? WTF

Reply:
The e2e tests should have caught this (if it's not already the case, can we make sure to add one that would). Unfortunately such an issue is not caught by the unit tests because of mocking.